tests/v2/validation/extensions/provisioning/permutations/permutations.go

```diff
@@ -0,0 +1,224 @@
+						require.NoError(s.T(), err)
+					// the custom cluster cases are all the same, in lue of the k8s version
+					case RKE1CustomCluster, RKE2CustomCluster, K3SCustomCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						//overriding rke2 kube version to be the same as k3s
+						testClusterConfig.KubernetesVersion = kubeVersion
+						clusterProviderSetup, err := CustomClusterSetup(clusterType)
+						if err != nil {
+							s.T().Fatal(err)
+						}
+						clusterName = clusterProviderSetup.ClusterCreationFunc(s.T(), client, *customProvider, testClusterConfig)
+					// airgap currently uses corral to create nodes and register with rancher
+					case RKE2AirgapCluster, K3SAirgapCluster, RKE1AirgapCluster:
+						clusterAirgapSetup, err := AirgapClusterEngineSetup(clusterType)
+						if err != nil {
+							s.T().Fatal(err)
+						}
+						clusterName = clusterAirgapSetup.ClusterCreationFunc(s.T(), client, testClusterConfig, corralPackages)
+					default:
+						clusterProviderSetup, err := CustomClusterSetup(clusterType)
+						if err != nil {
+							s.T().Fatal(err)
+						}
+						clusterName = clusterProviderSetup.ClusterCreationFunc(s.T(), client, *customProvider, testClusterConfig)
+					}
+					clusterNames = append(clusterNames, clusterName)
+					clusterConfigs = append(clusterConfigs, testClusterConfig)
+				})
+			}
+		}
+	}
+	logrus.Infof("Cluster Names: %v", clusterNames)
+	return clusterNames
+}
+
+func ConvertConfigToClusterConfig(clustersConfig *provisioning.Config) *provisioning.ClusterConfig {
+	var newConfig provisioning.ClusterConfig
+	newConfig.AddOnConfig = clustersConfig.AddOnConfig
+	newConfig.NodesAndRoles = &clustersConfig.NodesAndRoles
+	newConfig.NodesAndRolesRKE1 = &clustersConfig.NodesAndRolesRKE1
+	newConfig.AgentEnvVars = clustersConfig.AgentEnvVars
+	newConfig.Networking = clustersConfig.Networking
+	newConfig.Advanced = clustersConfig.Advanced
+	newConfig.Providers = &clustersConfig.Providers
+	newConfig.NodeProviders = &clustersConfig.NodeProviders
+	newConfig.ClusterAgent = clustersConfig.ClusterAgent
+	newConfig.FleetAgent = clustersConfig.FleetAgent
+	newConfig.Etcd = clustersConfig.Etcd
+	newConfig.LabelsAndAnnotations = clustersConfig.LabelsAndAnnotations
+	newConfig.Registries = clustersConfig.Registries
+	newConfig.UpgradeStrategy = clustersConfig.UpgradeStrategy
+
+	newConfig.Hardened = clustersConfig.Hardened
+	newConfig.PSACT = clustersConfig.Basics.PSACT
+	newConfig.PNI = clustersConfig.Basics.PNI
+
+	return &newConfig
+}
+
+func getClusterProvider(clusterType string, nodeProviderName string, clustersConfig *provisioning.Config) (*nodeProviders.Provider, *nodeProviders.RKE1Provider, *provisioning.ExternalNodeProvider, []string) {
+	var nodeProvider nodeProviders.Provider
+	var rke1NodeProvider nodeProviders.RKE1Provider
+	var customProvider provisioning.ExternalNodeProvider
+	var kubeVersions []string
+	switch clusterType {
```

Comment:
nit; let's add an extra line between the variable declarations and the switch statement.